cpp/src/arrow/compute/kernels/vector_replace_test.cc

```diff
@@ -792,5 +850,763 @@ TYPED_TEST(TestReplaceBinary, ReplaceWithMaskRandom) {
+  }
+}
+
+TYPED_TEST(TestFillNullNumeric, FillNullBackwardLargeInput) {
+  using CType = typename TypeTraits<TypeParam>::CType;
+  random::RandomArrayGenerator rand(/*seed=*/1000);
+  int64_t len_null = 500;
+  int64_t len_random = 1000;
+  std::shared_ptr<Array> array_random =
+      rand.ArrayOf(this->type(), len_random, /*nulls=*/0);
+  auto x_ptr = array_random->data()->template GetValues<CType>(1);
+  ASSERT_OK_AND_ASSIGN(auto array_null, MakeArrayOfNull(array_random->type(), len_null));
+  auto array_null_filled = ConstantArrayGenerator::Numeric<TypeParam>(len_null, x_ptr[0]);
+  {
+    ASSERT_OK_AND_ASSIGN(auto value_array,
+                         Concatenate({array_random, array_null, array_random}));
+    ASSERT_OK_AND_ASSIGN(auto result_array,
+                         Concatenate({array_random, array_null_filled, array_random}));
+    this->AssertFillNullArray(FillNullBackward, value_array, result_array);
+  }
+}
+
+TYPED_TEST(TestFillNullNumeric, FillNullForwardSliced) {
+  auto first_input_array =
+      this->array("[1, 4, null, null, 7, null, null, 8, 9, 5, null]");
+  auto expected_slice_length_2 =
+      this->array("[1, 4, null, null, 7, 7, null, 8, 9, 5, null]");
+  auto expected_slice_length_3 = this->array("[1, 4, 4, null, 7, 7, null, 8, 9, 5, 5]");
+  auto expected_slice_length_4 = this->array("[1, 4, 4, 4, 7, 7, 7, 8, 9, 5, 5]");
+  this->AssertFillNullArraySlices(FillNullForward, first_input_array,
+                                  expected_slice_length_2, expected_slice_length_3,
+                                  expected_slice_length_4);
+}
+
+TYPED_TEST(TestFillNullBinary, FillNullForwardSliced) {
+  auto first_input_array = this->array(
+      R"(["avb", "iyy", null, null, "mnh", null, null, "ttr", "gfd", "lkj", null])");
+  auto expected_slice_length_2 = this->array(
+      R"(["avb", "iyy", null, null, "mnh", "mnh", null, "ttr", "gfd", "lkj", null])");
+  auto expected_slice_length_3 = this->array(
+      R"(["avb", "iyy", "iyy", null, "mnh", "mnh", null, "ttr", "gfd", "lkj", "lkj"])");
+  auto expected_slice_length_4 = this->array(
+      R"(["avb", "iyy", "iyy", "iyy", "mnh", "mnh", "mnh", "ttr", "gfd", "lkj", "lkj"])");
+
+  this->AssertFillNullArraySlices(FillNullForward, first_input_array,
+                                  expected_slice_length_2, expected_slice_length_3,
+                                  expected_slice_length_4);
+}
+
+TYPED_TEST(TestFillNullNumeric, FillNullForwardRandomSliced) {
+  using CType = typename TypeTraits<TypeParam>::CType;
+  random::RandomArrayGenerator rand(/*seed=*/666);
+  int64_t len_random = 64;
+  int64_t len_null = 64;
+  int64_t slices = 15;
+  std::vector<std::shared_ptr<Array>> vector_values;
+  std::vector<std::shared_ptr<Array>> vector_filled;
+  for (int i = 0; i < slices; i++) {
+    std::shared_ptr<Array> array_random =
+        rand.Numeric<TypeParam>(len_random, /*min=*/5, /*max=*/200, /*null_count=*/0);
+    if (array_random) {
```

Comment:
Can these checks be eliminated?

That said, I'm not sure we need random tests for slicing. I think the hardcoded ones should be sufficient.

Reply:
Ok I deleted this random tests for slicing.